Runtimes/Supplemental/cmake/modules/FindSwiftCore.cmake

```
@@ -0,0 +1,120 @@
+#[=======================================================================[.rst:
+FindSwiftCore
+------------
+
+Find SwiftCore, deferring to SwiftCoreConfig.cmake when requested.
+This is meant to find the core library to be linked by the Supplemental libraries.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+The following :prop_tgt:`IMPORTED` TARGETS may be defined:
+
+ ``SwiftCore``
+
+Hint Variables
+^^^^^^^^^^^^^^
+
+ ``SDKROOT``
+   Set the path to the Swift SDK Root.
+   This only affects Windows builds.
+
+ ``Swift_SDKROOT``
+   Set the path to the Swift SDK installation.
+   This only affects Linux and Windows builds.
+   Apple builds always use the library provided by the SDK.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+The module may set the following variables if `SwiftCore_DIR` is not set.
+
+ ``SwiftCore_FOUND``
+   true if core was found
+
+ ``SwiftCore_INCLUDE_DIR``
+   the directory containing the Swift.swiftmodule folder
+
+#]=======================================================================]
+
+# If the SwiftCore_DIR_FLAG is specified, look there instead. The cmake-generated
+# config file is more accurate, but requires that the SDK has one available.
+if(SwiftCore_DIR)
+  if(SwiftCore_FIND_REQUIRED)
+    list(APPEND args REQUIRED)
+  endif()
+  if(SwiftCore_FIND_QUIETLY)
+    list(APPEND args QUIET)
+  endif()
+  find_package(SwiftCore NO_MODULE ${args})
+  return()
+endif()
+
+include(FindPackageHandleStandardArgs)
+
+if(APPLE)
+  # When building for Apple platforms, SwiftCore always comes from within the
+  # SDK as a tbd for a shared library in the shared cache.
+  find_path(SwiftCore_INCLUDE_DIR
+      "Swift.swiftmodule"
+    HINTS
+      "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  find_library(SwiftCore_IMPLIB
+    NAMES "libswiftCore.tbd"
+    HINTS
+      "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  add_library(SwiftCore SHARED IMPORTED GLOBAL)
+  set_target_properties(SwiftCore PROPERTIES
+    IMPORTED_IMPLIB "${SwiftCore_IMPLIB}")
```

I believe the target needs to have the path to the swiftmodules as well like its siblings:

```suggestion
    IMPORTED_IMPLIB "${SwiftCore_IMPLIB}"
    INTERFACE_INCLUDE_DIRECTORIES "${SwiftCore_INCLUDE_DIR}")
```